[FeedMergeBridge] Add new bridge

AntoineTurmel opened the pull request:
Here is a bridge that basically merge two feeds in one.
I'm not sure it's the best way, but it works for me.

Commit:
[FeedMergeBridge] Add new bridge

Here is a bridge that basically merge two feeds in one.
I'm not sure it's the best way, but it works for me.

Commit:
Update FeedMergeBridge.php

Commit:
Update FeedMergeBridge.php

logmanoriginal commented:
What is the use case for the bridge?
Wouldn't a feed aggregator be better suited for this?

AntoineTurmel commented:
Some usecases:
- Maximum feed limit for free plan on some aggregator services
- Custom single feed URL built with various content from other feeds (for integration)
- Integrate multiple bridges together: SteamBridge + SteamCommunityBridge / InstagramBridge + StoriesIGBridge...

Nick-Shannon commented:
I second this.  

Some widgets accept only one rss feed. 
I'm looking for a way to combine rss feeds into one.

AntoineTurmel commented:
> 
> 
> would it also be possible to make the amount of feeds to merge dynamic?

I don't think it's possible with the current parameters system to add dynamically more text fields...

em92 commented:
I suggest to fork rss-bridge and implement feed merging as a separate project.

Reasons:
1. Rss-bridge's goal and feed merging goal are different
2. You can make changes in code to add dynamic text inputs without considering current parameters system

AntoineTurmel commented:
As is it just a bridge, I'll keep it under a personal repository.
Forking could be a good idea, I'll look before if there are other alternatives in PHP first.

deathbybandaid commented:
Some Use Cases:

1) There are some feeds that I use the "Filter" bridge without filtering. This allows me to utilize the caching done via rss-bridge to multiple devices. **EDIT** I copied the Filter bridge and trimmed the filtering aspect out and made a cache bridge. (I'm not a php expert, so I'm sure there is a cleaner way)

2) Linus Media Groups on Youtube has multiple channels. It would be cool to be able to aggregate multiple channel feeds into 1 before it goes to my selfhosted Selfoss.

As for the goals of this bridge versus the projects being different,,, I'd personally see it as equivalent to the existing "filter" bridge.

teromene commented:
I am OK with this, as we already have multiple bridges that are not just transforming content, but I am listening if anyone is disagreeing,
Your thoughts, @LogMANOriginal ?

jlelse commented:
This looks like a really nice addition, I was looking for something like this :+1:

dvikan commented:
In my opinion we should try to keep this project pure with mostly bridges. We are not a feed reader.

dvikan commented:
Let's revisit this topic. I understand the scenarios where this is useful. And it doesn't hurt much to add one bridge that is a little bit special.

logmanoriginal commented:
Limiting the use of feeds by disallowing multiple feeds in a feed aggregator clearly defeats the purpose of an aggregator...

In my opinion this falls under the category of crippled APIs that rss-bridge is aimed at.

dvikan commented:
@AntoineTurmel I'm fine with merging this. Can you add proper example values and fix all linting issues etc?
http://lorem-rss.herokuapp.com/

Bockiii commented on bridges/FeedMergeBridge.php:
```diff
@@ -0,0 +1,59 @@
+					'type' => 'text',
+					'required' => true,
+					'title' => 'Insert your feed URL 1',
+					'exampleValue' => 'https://myfeed1.com'
```
```suggestion
					'exampleValue' => 'https://lorem-rss.herokuapp.com/feed?unit=day'
```

Bockiii commented on bridges/FeedMergeBridge.php:
```diff
@@ -0,0 +1,59 @@
+					'type' => 'text',
+					'required' => true,
+					'title' => 'Insert your feed URL2',
+					'exampleValue' => 'https://myfeed2.com'
```
```suggestion
					'exampleValue' => 'https://lorem-rss.herokuapp.com/feed?unit=year'
```

Bockiii reviewed (approved):
Added examplevalues

Bockiii committed:
Update bridges/FeedMergeBridge.php

examplevalue 1

Bockiii committed:
Update bridges/FeedMergeBridge.php

examplevalue 2

Bockiii commented:
There are some errors in the preview about the feedname. It looks like you want to merge the feednames but it's not doing it. Can you take a look at that? Thanks

dvikan commented:
@AntoineTurmel Ping

dvikan committed:
Test to see whether I can commit to PR branch

dvikan committed:
fix up bridge

dvikan committed:
Run phpcs . --standard=phpcs.xml --warning-severity=0 --extensions=php -p
............................................................  60 / 372 (16%)
............................................................ 120 / 372 (32%)
............................................................ 180 / 372 (48%)
............................................................ 240 / 372 (65%)
............................................................ 300 / 372 (81%)
..............................................E............. 360 / 372 (97%)
............                                                 372 / 372 (100%)

FILE: ...me/runner/work/rss-bridge/rss-bridge/bridges/FeedMergeBridge.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 8 | ERROR | String concat is not required here; use a single string
   |       | instead

dvikan committed:
style: lint

dvikan commented:
I fixed it up and merged it.

AntoineTurmel commented:
@dvikan I was busy with other things, thanks for fixing it :)
